CovidWatch/app/src/main/java/org/covidwatch/android/ble/BluetoothManager.kt

```diff
@@ -0,0 +1,181 @@
+package org.covidwatch.android.ble
+
+import android.app.*
+import android.content.ComponentName
+import android.content.Context
+import android.content.Context.BIND_AUTO_CREATE
+import android.content.Intent
+import android.content.ServiceConnection
+import android.os.Build
+import android.os.IBinder
+import android.util.Log
+import androidx.core.app.NotificationCompat
+import androidx.core.content.ContextCompat.getSystemService
+import org.covidwatch.android.MainActivity
+import org.covidwatch.android.R
+import org.covidwatch.android.data.ContactEvent
+import org.covidwatch.android.data.ContactEventDAO
+import org.covidwatch.android.data.CovidWatchDatabase
+import org.covidwatch.libcontactrace.BluetoothService
+import org.covidwatch.libcontactrace.BluetoothService.LocalBinder
+import org.covidwatch.libcontactrace.cen.*
+import org.covidwatch.libcontactrace.toBytes
+import org.covidwatch.libcontactrace.toUUID
+import java.util.*
+import java.util.concurrent.TimeUnit
+
+interface BluetoothManager {
+    fun startAdvertiser(cen: Cen)
+    fun stopAdvertiser()
+
+    fun startService(cen: Cen)
+    fun stopService()
+
+    fun changeAdvertisedValue(cen: Cen)
+}
+
+class BluetoothManagerImpl(
+    private val app: Application
+) : BluetoothManager {
+
+    private val intent get() = Intent(app, BluetoothService::class.java)
+
+    private var service: BluetoothService? = null
+
+    private val cenGenerator = DefaultCenGenerator()
+    private val cenVisitor = DefaultCenVisitor(app)
+    private var timer: Timer? = null
+
+    private val serviceConnection: ServiceConnection = object : ServiceConnection {
+        override fun onServiceConnected(name: ComponentName?, service: IBinder?) {
+            this@BluetoothManagerImpl.service = (service as LocalBinder).service.apply {
+                configure(
+                    BluetoothService.ServiceConfiguration(
+                        cenGenerator,
+                        cenVisitor,
+                        foregroundNotification()
+                    )
+                )
+                start()
+            }
+
+            runTimer()
+        }
+
+        override fun onServiceDisconnected(name: ComponentName?) = Unit
+    }
+
+    private fun runTimer() {
+        // scheduler a new timer to start changing the contact event numbers
+        timer?.cancel()
+        timer = Timer()
+        timer?.scheduleAtFixedRate(
+            object : TimerTask() {
+                override fun run() {
+                    service?.updateCen()
```

Comment:
@tzachari when I rewrote the plugin interface, I removed `updateCen`. It looks like we should actually be calling that from the timer as well (in addition to `get` so that the CENs actually change over time)